source/module_hamilt_lcao/module_gint/gint_force_cpu_interface.cpp

```diff
@@ -39,47 +40,110 @@ void Gint::cpu_force_interface(Gint_inout* inout) {
+        const ModuleBase::Array_Pool<double> psir_vlbr3 = Gint_Tools::get_psir_vlbr3(this->bxyz, na_grid, LD_pool, block_index, 
+                                                                                    cal_flag, vldr3, psir_ylm.get_ptr_2D());
+
+        ModuleBase::Array_Pool<double> psir_vlbr3_DM(this->bxyz, LD_pool);
+        ModuleBase::GlobalFunc::ZEROS(psir_vlbr3_DM.get_ptr_1D(), this->bxyz*LD_pool);
+
+	//calculating g_mu(r) = sum_nu rho_mu,nu f_nu(r)
+        if(GlobalV::GAMMA_ONLY_LOCAL)
+        {
+		//Gint_Tools::mult_psi_DM(*this->gridt, this->bxyz, na_grid, LD_pool, block_iw, block_size, block_index, cal_flag,
+		//	psir_vlbr3.get_ptr_2D(), psir_vlbr3_DM.get_ptr_2D(), DM_in, 2);
+            Gint_Tools::mult_psi_DM_new(
+                    *this->gridt, 
+                    this->bxyz, 
+                    grid_index, 
+                    na_grid, 
+                    LD_pool, 
+                    block_iw, 
+                    block_size, 
+                    block_index, 
+                    cal_flag,
+                    psir_vlbr3.get_ptr_2D(), 
+                    psir_vlbr3_DM.get_ptr_2D(), 
+                    this->DMRGint[inout->ispin], 
+                    false);
+        }
+        else
+        {
+            Gint_Tools::mult_psi_DMR(*this->gridt, this->bxyz, LD_pool, grid_index, na_grid, block_index, 
+                                        block_size, cal_flag, psir_vlbr3.get_ptr_2D(), psir_vlbr3_DM.get_ptr_2D(), 
+                                        this->DMRGint[inout->ispin], false);
+        }
+
+        if(inout->isforce)
+        {
+            //do integration to get force
+            this-> cal_meshball_force(grid_index, na_grid, block_size, block_index,psir_vlbr3_DM.get_ptr_2D(), 
+                                        dpsir_ylm_x.get_ptr_2D(), dpsir_ylm_y.get_ptr_2D(), dpsir_ylm_z.get_ptr_2D(),
+                                        fvl_dphi_thread);
+        }
+        if(inout->isstress)
+        {
+            //calculating g_mu(r)*(r-R) where R is the location of atom
+
+            // The array dpsirr contains derivatives of psir in the xx, xy, xz, yy, yz, zz directions,
+            // with each set of six numbers representing the derivatives in these respective directions.
+            ModuleBase::Array_Pool<double> dpsirr_ylm(this->bxyz, LD_pool * 6);
+            Gint_Tools::cal_dpsirr_ylm(*this->gridt, this->bxyz, na_grid, grid_index, block_index, 
+                                        block_size, cal_flag,dpsir_ylm_x.get_ptr_2D(), dpsir_ylm_y.get_ptr_2D(),
+                                        dpsir_ylm_z.get_ptr_2D(),dpsirr_ylm.get_ptr_2D());
+
+            //do integration to get stress
+            this-> cal_meshball_stress(na_grid, block_index, psir_vlbr3_DM.get_ptr_1D(), 
+                                        dpsirr_ylm.get_ptr_1D(), svl_dphi_thread);
+        }
+
+    //release memories
+        delete[] block_iw;
```

Comment:
stop using new and delete for these variables

Reply:
The modifications have been completed, including vindex, vldr3, block_iw, block_size, block_index. Meanwhile, cal_flag has been refactored using Array_Pool, and comments in Doxygen format have been added.